engine/runtime/src/main/java/org/enso/interpreter/runtime/TruffleCompilerContext.java

```diff
@@ -589,8 +588,8 @@ private boolean doSerializeLibrarySuggestions(
   }
 
   @Override
-  public scala.Option<List<org.enso.polyglot.Suggestion>> deserializeSuggestions(
-      LibraryName libraryName) throws InterruptedException {
+  public scala.Option<Object> deserializeSuggestions(LibraryName libraryName)
```

Comment:
Could have probably parameterized `TruffleCompilerContext` and use an abstract type parameter here, instead of an `Object` (and casting later). If `TruffleCompilerContext` wasn't final of course.

Reply:
The problem with adding a type parameter is that it is transitive. It propagates to all usage places. If `CompilerContext` was parametrized, then also `Compiler` needs to be parametrized and all usages need to be parametrized. Right now I am trying to avoid `TruffleFile` in `PackageRepository` - parametrizing the type with `[F]` is an option, but the diff is massive... I am trying to avoid it.